docs/parameters.md

```diff
@@ -1 +1,255 @@
+
+This approach has several drawbacks:
+
+- It's not guaranteed to be safe from SQL injection
+- The server could not boost performance by caching prepared statements
+- The parameter marker syntax conflicted with SQL syntax in some cases
+
+Nevertheless, this behaviour is preserved in version 3.0.0 and above for legacy purposes. It will be removed in a subsequent major release. To enable this legacy code path, you must now construct your connection with `use_inline_params=True`.
+
+## Requirements
+
+Rendering parameters inline is supported on all versions of DBR since these queries are indistinguishable from ad-hoc query text.
+
+
+## SQL Syntax
+
+Variables in your SQL query can look like `%(param)s` or like `%s`. 
+
+#### Example
+
+```sql
+-- pyformat paramstyle is used for named parameters
+SELECT * FROM table WHERE field = %(value)s
+
+-- %s is used for positional parameters
+SELECT * FROM table WHERE field = %s
+```
+
+## Python Syntax
+
+This connector follows the [PEP-249 interface](https://peps.python.org/pep-0249/#id20). The expected structure of the parameter collection follows the paramstyle of the variables in your query. 
+
+### `pyformat` paramstyle Usage Example
+
+Parameters must be passed as a dictionary.
+
+```python
+from databricks import sql
+
+with sql.connect(..., use_inline_params=True) as conn:
+    with conn.cursor() as cursor():
+        query = "SELECT field FROM table WHERE field = %(value1)s AND another_field = %(value2)s"
+        parameters = {"value1": "foo", "value2": 20}
+        result = cursor.execute(query, parameters=parameters).fetchone()
+```
+
+The above query would be rendered into the following SQL:
+
+```sql
+SELECT field FROM table WHERE field = 'foo' AND another_field = 20
+```
+
+### `%s` paramstyle Usage Example
+
+Parameters must be passed as a list.
+
+```python
+from databricks import sql
+
+with sql.connect(..., use_inline_params=True) as conn:
+    with conn.cursor() as cursor():
+        query = "SELECT field FROM table WHERE field = %s AND another_field = %s"
+        parameters = ["foo", 20]
+        result = cursor.execute(query, parameters=parameters).fetchone()
+```
+
+The result of the above two examples is identical.
+
+**Note**: `%s` is not compliant with PEP-249 and only works due to the specific implementation of our inline renderer. 
+
+**Note:** This `%s` syntax overlaps with valid SQL syntax around the usage of `LIKE` DML. For example if your query includes a clause like `WHERE field LIKE '%sequence'`, the parameter inlining function will raise an exception because this string appears to include an inline marker but none is provided.
```

Comment:
you should explain in this note that if they switch to using named parameters, we will pass this along to the database, allowing it to be used as expected in the LIKE statements.  This might be inferred from the other documentation, but better to be explicit here.

Reply:
Added!